### PR TITLE
#1 - Constrained systems

### PR DESCRIPTION
**Pull request by mforets**

### CODE REVIEW

**Comment by mforets** [comment]
note that `LazySet` is *not* required, but i found it convenient for writing tests. we can just update `.travis.yml` , "after success" to install `LazySet`.

**Comment by schillic** [comment]
I am fine with using `LazySets` in the tests. It should be intuitive enough to read.

**Comment by mforets** [comment]
the `osx` Travis build makes a big slow down (Running for 1 hr 36 min 38 sec), while it's < 3min for linux. then i've removed, please undo if u really want this or know why it's so slow -- see also [osx builds extremely slow to start](https://github.com/travis-ci/travis-ci/issues/7304)

File: src/continuous.jl [hunk 1]
```diff
@@ -10,9 +10,83 @@ x' = A x.
...
+
+- `A` -- square matrix
+- `B` -- matrix
+"""
+struct LinearControlContinuousSystem{T, MT <: AbstractMatrix{T}} <: AbstractContinuousSystem
+    A::MT
+    B::MT
+end
+function LinearControlContinuousSystem{T, MT <: AbstractMatrix{T}}(A::MT, B::MT)
+    @assert size(A, 1) == size(B, 1)
```
Comment (blegat) [inline src/continuous.jl]: We could check here that A is square by doing `@assert Base.LinAlg.checksquare(s.A) == size(B, 1)` instead

File: src/continuous.jl [hunk 2]
```diff
@@ -10,9 +10,83 @@ x' = A x.
...
+- `U` -- input constraints
+"""
+struct ConstrainedLinearControlContinuousSystem{T, MT <: AbstractMatrix{T}, ST, UT} <: AbstractContinuousSystem
+    A::MT
+    B::MT
+    X::ST
+    U::UT
+end
+function ConstrainedLinearControlContinuousSystem{T, MT <: AbstractMatrix{T}, ST, UT}(A::MT, B::MT, X::ST, U::UT)
+    @assert size(A, 1) == size(B, 1)
```
Comment (blegat) [inline src/continuous.jl]: same here

File: test/runtests.jl [hunk 3]
```diff
@@ -1,4 +1,4 @@
-using Systems
+using Systems, LazySets
```
Comment (blegat) [inline test/runtests.jl]: Why do we need LazySets ?

Comment (schillic) [reply test/runtests.jl]: > note that LazySet is not required, but i found it convenient for writing tests. we can just update .travis.yml , "after success" to install LazySet.

Comment (blegat) [reply test/runtests.jl]: Yes, I just noticed that comment ^ ^

**Comment by blegat** [comment]
It seems to me that Continuous Control System is more common than Control Continuous System. I have done a quick google search. If I use either term, the first result is : https://link.springer.com/chapter/10.1007/978-1-84628-056-6_1 which uses "Continuous Control System".

**Comment by schillic** [comment]
> It seems to me that Continuous Control System is more common than Control Continuous System.

@mforets probably chose this because of [my suggestion here](https://github.com/JuliaReach/Systems.jl/issues/1#issuecomment-352567896). I am of course fine with breaking this rule of thumb for existing terms.

**Comment by mforets** [comment]
it's true that common usage in this context is "continuous-time systems" or "discrete-time systems", which are then specialized into more concrete classes. i'm favorable of accepting this slight abuse of notation and use "xyzContinuousSystem" and similarly for discrete, to have some homogeneity in the naming convention.

... we could go further to write "xyzCS" and "xyzDS" respectively but that's kind of discouraged in Julia.

on a separate note, thanks for the suggestions to improve the constructors, which i've incorporated. let me continue by adding the discrete analogues to this branch.

**Pull request merged**